### ISSUE TITLE
Content is reset at unexpected times

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/learn.svelte.dev#298]
I'm sure there's a simpler repro than this, but I'm making a note of what I observed just now for further investigation later.

1. Go to https://learn.svelte.dev/tutorial/textarea-inputs
2. Hit 'solve'
3. Change `- lists` to `- [links](https://example.com)`
4. Click the newly-rendered link in the `output` section
5. Close the tab

The content will have been reset (this happens for me in Firefox and Chrome).

Weirdly, you have to actually close the tab — if you leave it open and navigate back to the tutorial, the content won't have been reset. And if you drag the newly opened tab into a separate window and _then_ close it, nothing will happen. It's... strange.

**Comment by tomoam** [comment]
I found that the iframe was reloading.

https://user-images.githubusercontent.com/29677552/227553967-32d3c1bf-6e21-45b1-821e-287595f9b2ee.mov

---

The relevant code is here.

https://github.com/sveltejs/learn.svelte.dev/blob/6736ff71af56c3c6cada90e0ad13f299e387d962/content/tutorial/common/src/__client.js#L112-L124

https://github.com/sveltejs/learn.svelte.dev/blob/6736ff71af56c3c6cada90e0ad13f299e387d962/src/routes/tutorial/%5Bslug%5D/Output.svelte#L54-L62

---

This is caused by throttling of setInterval and setTimeout.

https://developer.chrome.com/blog/timer-throttling-in-chrome-88/

The `ping` in `__client.js` runs every 100ms with `setInterval` and sends the message. The `handle_message` in `Output.svelte` receives that message. If the message does not come for more than 1000 milliseconds, the iframe is reloaded.

However, when the tab goes to the background, `setInterval` and `setTimeout` are throttled to run approximately every 1000 - 1500 milliseconds. This means that sometimes the `setTimeout` delay time in the `handle_message` is shorter than the `ping` delay time, at which time `set_iframe_src` is called and the iframe is reloaded.

I have found the cause, but I don't know how to fix it.
